game-core/src/main/java/games/strategy/triplea/delegate/power/calculator/AaDefenseCombatValue.java

```diff
@@ -0,0 +1,67 @@
+package games.strategy.triplea.delegate.power.calculator;
+
+import games.strategy.engine.data.GameData;
+import games.strategy.engine.data.Unit;
+import games.strategy.triplea.attachments.UnitSupportAttachment;
+import java.util.function.Predicate;
+import lombok.AccessLevel;
+import lombok.Builder;
+import lombok.EqualsAndHashCode;
+import lombok.Getter;
+import lombok.Value;
+
+/**
+ * Calculates defense strength and roll for AA/Targeted dice
+ *
+ * <p>This takes into account friendly support, and enemy support
+ */
+@Value
+@EqualsAndHashCode(callSuper = true)
+@Getter(AccessLevel.NONE)
+class AaDefenseCombatValue extends AaCombatValue {
+
+  @Builder
+  AaDefenseCombatValue(
+      final GameData gameData,
+      final AvailableSupports supportFromFriends,
+      final AvailableSupports supportFromEnemies) {
+    super(gameData, supportFromFriends, supportFromEnemies);
+  }
+
+  @Override
+  public StrengthAndRollCalculator getStrength() {
+    return new AaDefenseStrength(gameData, supportFromFriends, supportFromEnemies);
+  }
+
+  @Override
+  public boolean isDefending() {
+    return true;
+  }
+
+  static class AaDefenseStrength extends StrengthAndRollCalculator {
+
+    private final GameData gameData;
+
+    AaDefenseStrength(
+        final GameData gameData,
+        final AvailableSupports friendlySupport,
+        final AvailableSupports enemySupport) {
+      super(friendlySupport, enemySupport);
+      this.gameData = gameData;
+    }
+
+    @Override
+    public int getValue(final Unit unit) {
```

Comment:
For future consideration, a more specific name than 'getValue' could be of benefit. Specifically to consider, what kind of value are we returning? Is this the roll value, the total strength, or just the die-strength?